incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -38,33 +45,50 @@ type ConfigReconciler struct {
 // Reconcile sets up some basic variable and logs the Spec.
 // TODO: Updates the comment above when adding more logic to the Reconcile method.
 func (r *ConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	// TODO: Surface the error more prominently and add a validating admission controller to prevent
-	// the problem in the first place.
+	ctx := context.Background()
```

Comment:
We've tried to keep most functions fairly short by aggressively moving code to subfunctions.  Maybe try to do the same here?

Reply:
Good idea, thanks for the suggestion. PTAL.